dev/type_traits.h

```diff
@@ -58,4 +61,9 @@ namespace sqlite_orm {
         template<typename T>
         using on_type_t = typename T::on_type;
     }
+
+#ifdef SQLITE_ORM_CPP20_CONCEPTS_SUPPORTED
+    template<class T>
+    concept orm_names_type = requires { typename T::type; };
```

Comment:
I think it should remain in the public namespace because it is part of the public requirements [`orm_column_alias`](https://github.com/FireDaemon/sqlite_orm/blob/d87938ef9b48204465030466b10955bd1b7dbb7f/dev/alias_traits.h#L66) and [`orm_recordset_alias`](https://github.com/FireDaemon/sqlite_orm/blob/d87938ef9b48204465030466b10955bd1b7dbb7f/dev/alias_traits.h#L75).